### PR TITLE
Added grunt based development setup template

### PR DESCRIPTION
**Pull request by nitish-mehta**
- Added grunt based setup
- Updated app files to reflect correct structure
- Conditional copy of files

### CODE REVIEW

File: app/index.js [hunk 1]
```diff
@@ -49,52 +51,115 @@ module.exports = class extends Generator {
 		const oProps = {
 			libraryName: sLibName,
 			controlName: sControlName,
-			dir: sDir,
+			baseDir: sDir,
+			srcCodeDir: `src/${sDir}`,
 			libraryNameOnly: sLibNameOnly,
+			gitRepository: '',
```
Comment (vaibhavar) [inline app/index.js]: TODO - get from input parameters

File: app/index.js [hunk 2]
```diff
@@ -49,52 +51,115 @@ module.exports = class extends Generator {
...
 		this.log('\n\n Buzzing the engines');
+
+		// create library
+		this._createTemplateLibrary(oProps);
+
+		// create development setup
+		this._createDevelopmentSetup(oProps);
+
+		// install required npm packages
+		// TODO
```
Comment (vaibhavar) [inline app/index.js]: remove TODO

File: app/templates/devresources/.editorconfig [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# see http://editorconfig.org
```
Comment (vaibhavar) [inline app/templates/devresources/.editorconfig]: is this required ?

File: app/index.js (hunk 1 above)
Comment (nitish-mehta) [reply app/index.js]: Done.

File: app/templates/devresources/.editorconfig (hunk 3 above)
Comment (nitish-mehta) [reply app/templates/devresources/.editorconfig]: Yes. This helps with consistent code styling across editors.

File: app/index.js (hunk 2 above)
Comment (nitish-mehta) [reply app/index.js]: Done.

**Pull request merged**